File: test-suite/src/test/java/io/micronaut/kafka/docs/quickstart/RuntimeBootstrapServers.java
package io.micronaut.kafka.docs.quickstart;

// tag::imports[]
import io.micronaut.context.ApplicationContextBuilder;
import io.micronaut.context.ApplicationContextConfigurer;
import io.micronaut.context.annotation.ContextConfigurer;

import java.util.Map;
// end::imports[]

// tag::clazz[]
@ContextConfigurer
public final class RuntimeBootstrapServers implements ApplicationContextConfigurer {
    @Override
    public void configure(ApplicationContextBuilder builder) {
        builder.properties(Map.of(
            "kafka.bootstrap.servers", resolveBootstrapServers()
        ));
    }

    private static String resolveBootstrapServers() {
        return "localhost:9092";
    }
}
// end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/kafka/docs/quickstart/RuntimeBootstrapServers.kt
package io.micronaut.kafka.docs.quickstart

// tag::imports[]
import io.micronaut.context.ApplicationContextBuilder
import io.micronaut.context.ApplicationContextConfigurer
import io.micronaut.context.annotation.ContextConfigurer
// end::imports[]

// tag::clazz[]
@ContextConfigurer
class RuntimeBootstrapServers : ApplicationContextConfigurer {
    override fun configure(builder: ApplicationContextBuilder) {
        builder.properties(
            mapOf(
                "kafka.bootstrap.servers" to resolveBootstrapServers()
            )
        )
    }
// end::clazz[]

    private fun resolveBootstrapServers(): String = "localhost:9092"
}
